src/contributing.md

```diff
@@ -119,6 +119,24 @@ when contributing to Rust under [the git section](./git.md).
 [about-pull-requests]: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests
 [development-models]: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/getting-started/about-collaborative-development-models#fork-and-pull-model
 
+### Separate your changes and commit guidelines
+
+A PR might include various commits. For example, if your change includes both bug-fixes and
+performance enhancements for a single module. Please separate those changes into two or more
+individual commits.
+
+On the other hand, if your change encompasses numerous files, include those files in your commit.
+**A single logical change is contained within a single patch**. The commit message should describe
+what you are changing (e.g. `Binary` trait bug-fixes in the `fmt` module). Even if another commit
+in the same PR already says it. **Each commit message should be its own unit of information.**
```

Comment:
hm, the example message feels too general... I would expect it to be a PR title, instead of a commit message